frontend/src/metabase/query_builder/components/view/ConvertQueryModal/utils.ts

```diff
@@ -0,0 +1,20 @@
+import { DatasetQuery } from "metabase-types/api";
+import { checkNotNull } from "metabase/core/utils/types";
+import Question from "metabase-lib/Question";
+
+export function createDatasetQuery(
+  query: string,
+  question: Question,
+): DatasetQuery {
+  const tableId = question.tableId();
+  const collection =
+    tableId === null || typeof tableId === "undefined"
```

Comment:
just `tableId != null`, this is a well-known pattern used across the codebase. It will check for both `null` and `undefined`